src/routers/secure/debug.py

```
@@ -0,0 +1,631 @@
+    remaining_count: int
+
+
+@router.get("/network-stats", operation_id="get_network_stats")
+async def get_network_stats() -> NetworkStatsResponse:
+    """Get comprehensive network profiling statistics."""
+    try:
+        from program.utils.network_profiler import network_profiler
+        stats = network_profiler.get_statistics()
+        return NetworkStatsResponse(**stats)
+    except ImportError:
+        raise HTTPException(status_code=503, detail="Network profiling not available")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)